soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,1046 @@
+    gRandomizeShuffleBeans = RO_GENERIC_YES;
+    gRandomizeShuffleScrubs = RO_GENERIC_YES;
+    gRandomizeShuffleMerchants = RO_GENERIC_YES;
+    gRandomizeShuffleCows = RO_GENERIC_NO;
+    gRandomizeShuffleAdultTrade = RO_GENERIC_YES;
+    gRandomizeShuffleKokiriSword = RO_GENERIC_YES;
+    gRandomizeShuffleWeirdEgg = RO_GENERIC_YES;
+    gRandomizeShuffleGerudoCard = RO_GENERIC_YES;
+    gRandomizeShuffleFrogSongRupees = RO_GENERIC_NO;
+    gRandomizeStartingMapsCompasses = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeKeysanity = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeGerudoFortressKeys = RO_GERUDO_KEYS_VANILLA;
+    gRandomizeBossKeysanity = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeShuffleGanonBossKey = RO_GANON_BOSS_KEY_ANYWHERE;
+    gRandomizeGerudoFortress = RO_GF_NORMAL;
+}
+
+void SetRandomizerSettings() {
+    gRandomizeShopsanity =              OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHOPSANITY);
+    gRandomizeShuffleTokens =           OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_TOKENS);
+    gRandomizeShuffleBeans =            OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_MAGIC_BEANS);
+    gRandomizeShuffleScrubs =           OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_SCRUBS);
+    gRandomizeShuffleMerchants =        OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_MERCHANTS);
+    gRandomizeShuffleCows =             OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_COWS);
+    gRandomizeShuffleAdultTrade =       OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_ADULT_TRADE);
+    gRandomizeShuffleKokiriSword =      OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_KOKIRI_SWORD);
+    gRandomizeShuffleWeirdEgg =         OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_WEIRD_EGG);
+    gRandomizeShuffleGerudoCard =       OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_GERUDO_MEMBERSHIP_CARD);
+    gRandomizeShuffleFrogSongRupees =   OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHUFFLE_FROG_SONG_RUPEES);
+    gRandomizeStartingMapsCompasses =   OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_STARTING_MAPS_COMPASSES);
+    gRandomizeKeysanity =               OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_KEYSANITY);
+    gRandomizeGerudoFortressKeys =      OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_GERUDO_KEYS);
+    gRandomizeBossKeysanity =           OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_BOSS_KEYSANITY);
+    gRandomizeShuffleGanonBossKey =     OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_GANONS_BOSS_KEY);
+    gRandomizeGerudoFortress =          OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_GERUDO_FORTRESS);
+}
+
+bool IsVisibleInCheckTracker(RandomizerCheckObject rcObj) {
+    return
```

Comment:
instead of having this as a method, you could just have a static `isInitialized` bool in `IsVisibleInCheckTracker` and then move the reading RSK logic into an `if` in there

if you want to get really fancy with it, you could simplify the checks in `IsVisibleInCheckTracker` a bit. for example, instead of having
```cpp
gRandomizeShopsanity = RO_SHOPSANITY_OFF;
```
```cpp
gRandomizeShopsanity =              OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHOPSANITY);
```
```cpp
(rcObj.rcType != RCTYPE_SHOP || gRandomizeShopsanity > RO_SHOPSANITY_ZERO_ITEMS) &&
```
you could have
```cpp
static bool shouldShowShopsanity = gSaveContext.n64ddFlag ? (
    OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHOPSANITY) != RO_SHOPSANITY_OFF &&
    OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_SHOPSANITY) != RO_SHOPSANITY_ZERO_ITEMS) : false;
```


```c
rcObj.rcType != RCTYPE_SHOP || shouldShowShopsanity
```

Reply:
The variable name structure was a holdover from when it was originally using the CVars to get the settings (which isn't true, it has to use the settings from the save file), and it was just a direct match to the CVar string to simplify remembering. And the layout of code was a pre-optimization to avoid having to always get the rando settings. 

IMO, having it separate makes it more readable, but I can try to refactor to 1 or 2 methods that handle all 4 blocks of code, and throw some comments for the readability.

Reply:
if you want to just have `IsVisibleInCheckTracker` that calls `IsVisibleInVanillaCheckTracker` or `IsVisibleInRandoCheckTracker` or something that could work, my main concern from a readability stand point was what felt like hacky use of randomizer settings values to achieve desired behavior in vanilla

Reply:
Refactored for readability with 69719abf7054f0d0fba3a3b8c55f94f414c9db7e

Reply:
couple of things i noticed in the logic (out of scope for this PR)
* skulltulas will appear on the tracker when
  * playing vanilla
  * rando'd with tokens shuffled
this means the tracker won't work for going through to find any missing tokens when not playing tokensanity

fortress key checks won't appear in vanilla, because `fortressNormal` is set to false

this: `//Bug: the above will spoil that everything is vanilla if the random count rolled 0.`

 it'd probably be good to make an issue documenting this stuff so it can be addressed in a follow up PR at some point

Reply:
@briaguya-ai Skulltullas appearing in vanilla is actually intentional, as my goal for vanilla thus far has been to show all checks that are required for the completionists out there who want to 100% the game.

Reply:
@sonoftunk 
> Skulltullas appearing in vanilla is actually intentional

yes, that makes sense, the part that doesn't make sense is the lack of ability to track skull token locations in rando when not playing with tokensanity (this is handled in other trackers by having a toggle for show/hide skulltulas)